src/mono/wasm/test-main.js

```diff
@@ -105,7 +105,27 @@ function set_exit_code(exit_code, reason) {
         }
 
     } else if (App && App.INTERNAL) {
-        App.INTERNAL.mono_wasm_exit(exit_code);
+        if (is_node) {
+            let _flush = function(_stream) {
+                return new Promise((resolve, reject) => {
+                    if (!_stream.write('')) {
+                        _stream.on('drain', () => resolve());
+                        setTimeout(reject, 3000);
+                    } else {
+                        resolve();
+                    }
+                });
+            };
+            let stderrFlushed = _flush(process.stderr);
+            let stdoutFlushed = _flush(process.stdout);
+
+            Promise.all([ stdoutFlushed, stderrFlushed ])
+                    .then(
+                        () => App.INTERNAL.mono_wasm_exit(exit_code),
```

Comment:
one of these is the rejection handler, right? if draining fails should we make sure the exit code is nonzero and log the error?

Reply:
but .. how do we ensure that the new log message gets flushed? Or maybe we can ignore that potential issue, because the tests would fail with the non-zero exit code, and the dev can then try to debug?

Reply:
We can't, but I think it's better than exit code 0